io/driver/driver_parameter.py

```diff
@@ -0,0 +1,170 @@
+                                     interface_type='Ethernet')
+        self.load_unload_sleep_time = 30
+        self.error_modules = []
+        self.mod_list = []
+        self.uname = linux_modules.platform.uname()[2]
+
+    def built_in_module(self, module):
+        """
+        checking whether the given module is built_in module or not
+        """
+        path = "/lib/modules/%s/modules.builtin" % self.uname
+        for each in genio.read_all_lines(path):
+            out = process.getoutput(each.split('/')[-1])
+            if module == out.split('.'[0]):
+                return True
+            return False
+
+    def ping_check(self, options):
+        '''
+        Checks if the ping to peer works. Returns True if it works.
+        Returns False otherwise.
+        '''
+        cmd = "ping -I %s %s %s" % (self.ifaces, options, self.peer)
+        if process.system(cmd, shell=True, verbose=True,
+                          ignore_status=True) != 0:
+            return False
+        return True
+
+    def sysfs_value_check(self):
+        '''
+        Checks if sysfs value matches to test value, Returns True if yes.
+        Returns False otherwise.
+        '''
+        value_check = process.system_output('/usr/bin/cat \
+                               /sys/module/%s/parameters/%s' %
+                                            (self.module, self.param_name
+                                             )).decode('utf-8')
+        if self.param_value not in value_check:
+            return False
+        return True
+
+    def param_check(self):
+        '''
+        Checks if Param is available for the Module, Returns True if yes.
+        Returns False otherwise.
+        '''
+        value_check = process.system_output('/usr/sbin/modinfo -p %s '
+                                            % self.module).decode('utf-8')
+        if self.param_name not in value_check:
+            return False
+        return True
+
+    def module_load_unload(self, mod1):
+        """
+        Unloading and loading the given module
+        """
+        if linux_modules.module_is_loaded(mod1) is False:
+            linux_modules.load_module(mod1)
+            time.sleep(self.load_unload_sleep_time)
+        sub_mod = linux_modules.get_submodules(mod1)
+        if sub_mod:
+            for mod in sub_mod.split(' '):
+                linux_modules.unload_module(mod)
+                if linux_modules.module_is_loaded(mod) is True:
+                    self.error_modules.append(mod)
+                    break
+        cmd = "rmmod %s" % mod1
+        if process.system(cmd, shell=True, verbose=True,
+                          ignore_status=True) != 0:
+            self.fail("Unloading Module %s failed" % mod1)
+        time.sleep(self.load_unload_sleep_time)
+        location = process.system_output('/sbin/modinfo -n %s' %
+                                         self.module).decode('utf-8')
+        cmd1 = "insmod %s %s=%s" % \
```

Comment:
Please use https://github.com/avocado-framework/avocado/blob/master/avocado/utils/linux_modules.py#L46

Reply:
explaination same as above^^ ...insmod has to be followed by rmmod.    Enhancing exsisting function might break the other scripts.

@narasimhan-v

Reply:
Changes included in latest code!!